traceapp/app.go

```diff
@@ -239,11 +285,15 @@ func (a *App) serveTraceUpload(w http.ResponseWriter, r *http.Request) error {
 	if err != nil {
 		return err
 	}
+	return a.uploadTraces(traces...)
+}
 
+// uploadTraces uploads literal traces into the storage system for later viewing.
+func (a *App) uploadTraces(traces ...*appdash.Trace) error {
 	// Collect the unmarshaled traces, ignoring any previously existing ones (i.e.
```

Comment:
Correct, this exact case is handled by the first `a.Store.Trace` call in the for loop below.